MBC_RegistrationMobile.class.inc

```diff
@@ -63,12 +63,8 @@ class MBC_RegistrationMobile
     $payloadDetails = unserialize($payload->body);
     $deliveryTag = $payload->delivery_info['delivery_tag'];
 
-    // @todo - remove $payloadDetails['activity'] check when this consumer handles all Mobile Commons signups
-    if ((isset($payloadDetails['mobile']) && $payloadDetails['mobile'] != '') &&
-        ($payloadDetails['activity'] == 'user_welcome-niche' ||
-         $payloadDetails['activity'] == 'user_import' ||
-         $payloadDetails['activity'] == 'cgg2014_signup' ||
-         $payloadDetails['activity'] == 'cgg2014_vote')) {
+    if (!empty($payloadDetails['mobile']) &&
```

Comment:
Basically it checks that the variable is set and filled in with data.
http://php.net/manual/en/function.empty.php

> The following things are considered to be empty:
> - "" (an empty string)
> - 0 (0 as an integer)
> - 0.0 (0 as a float)
> - "0" (0 as a string)
> - NULL
> - FALSE
> - array() (an empty array)
> - $var; (a variable declared, but without a value)